Terminal.Gui/Views/ComboBox.cs

```diff
@@ -185,15 +185,19 @@ public IListDataSource Source
             // Only need to refresh list if its been added to a container view
             if (SuperView is { } && SuperView.Subviews.Contains (this))
             {
-                SelectedItem = -1;
-                _search.Text = string.Empty;
-                Search_Changed (this, new CancelEventArgs<string> (string.Empty, _search.Text));
+                Text = string.Empty;
+//                SelectedItem = -1;
+//                _search.Text = string.Empty;
+//                ResetSearchSet ();
+
+//                HideList ();
+////                ShowHideList (string.Empty);
                 SetNeedsDisplay ();
             }
         }
     }
 
-    /// <summary>The currently selected list item</summary>
+    /// <summary>The text of the currently selected list item</summary>
     public new string Text
```

Comment:
Why `new`?

Reply:
`ComboBox` does a lot of weird things like this. It's quite fragile.

One of the reasons I suggested this:

- #2404

I'm not touching it any more than I already have because I'm terrified of breaking it.